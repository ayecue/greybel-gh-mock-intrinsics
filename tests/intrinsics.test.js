const {
  Debugger,
  OutputHandler,
  HandlerContainer,
  ObjectValue
} = require('greybel-interpreter');
const {
  Interpreter
} = require('greyscript-interpreter');
const defaultInit = require('greybel-intrinsics').init;
const { init, createGHMockEnv } = require('../dist');
const fs = require('fs');
const path = require('path');
const testFolder = path.resolve(__dirname, 'scripts');

let printMock;

const testDate = new Date(1642924301240);

Date.now = () => testDate.getTime();

class TestOutputHandler extends OutputHandler {
  print(vm, value) {
    printMock(value);
  }

  update(vm, value) {
    printMock(value);
  }

  progress(vm, time) {
    return Promise.resolve();
  }

  waitForInput(vm, isPassword, value) {
    printMock(value);
    return Promise.resolve('test');
  }

  waitForKeyPress(vms, value) {
    printMock(value);
    return Promise.resolve({
      keyCode: 13,
      code: 'Enter'
    });
  }

  clear() {
    printMock('clearing screen');
  }
}

class TestDebugger extends Debugger {
  debug() {
    printMock("debug");
  }

  interact(vm, item) {
    printMock("interact at " + item.start.line);
  }

  resume() {
    printMock("resume");
  }
}

describe('interpreter', function () {
  beforeEach(function () {
    printMock = jest.fn();
  });

  describe('default scripts', function () {
    fs.readdirSync(testFolder).forEach((file) => {
      const filepath = path.resolve(testFolder, file);

      test(path.basename(filepath), async () => {
        const interpreter = new Interpreter({
          target: filepath,
          handler: new HandlerContainer({
            outputHandler: new TestOutputHandler()
          }),
          debugger: new TestDebugger()
        });
        let success = false;

        interpreter.setApi(defaultInit(init(new ObjectValue(), createGHMockEnv(interpreter, {
          myProgramContent: fs.readFileSync(filepath, {
            encoding: 'utf-8'
          })
        }))));

        try {
          await interpreter.run();
          success = true;
        } catch (e) {
          console.log(`${filepath} failed with: `, e);
        }

        expect(success).toEqual(true);
        for (const call of printMock.mock.calls) {
          expect(call[0]).toMatchSnapshot();
        }
      });
    });
  });
});
