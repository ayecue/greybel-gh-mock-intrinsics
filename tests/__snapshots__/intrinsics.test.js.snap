// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interpreter default scripts blockchain.src 1`] = `"blockchainLib"`;

exports[`interpreter default scripts blockchain.src 2`] = `"blockchainLib"`;

exports[`interpreter default scripts blockchain.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts blockchain.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts blockchain.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts blockchain.src 6`] = `"coin"`;

exports[`interpreter default scripts blockchain.src 7`] = `"wallet"`;

exports[`interpreter default scripts blockchain.src 8`] = `"wallet"`;

exports[`interpreter default scripts blockchain.src 9`] = `"Not yet supported"`;

exports[`interpreter default scripts class-output.src 1`] = `"test"`;

exports[`interpreter default scripts class-output.src 2`] = `"{\\"__isa\\":test}"`;

exports[`interpreter default scripts class-output.src 3`] = `"{\\"__isa\\":test,\\"foo\\":123,\\"bar\\":0}"`;

exports[`interpreter default scripts class-output.src 4`] = `"test"`;

exports[`interpreter default scripts class-output.src 5`] = `"map"`;

exports[`interpreter default scripts class-output.src 6`] = `"map"`;

exports[`interpreter default scripts coin.src 1`] = `"coin"`;

exports[`interpreter default scripts coin.src 2`] = `"coin"`;

exports[`interpreter default scripts coin.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 6`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 7`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 8`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 9`] = `"subwallet"`;

exports[`interpreter default scripts coin.src 10`] = `"[subwallet]"`;

exports[`interpreter default scripts coin.src 11`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 12`] = `"Not yet supported"`;

exports[`interpreter default scripts coin.src 13`] = `"Not yet supported"`;

exports[`interpreter default scripts crypto.src 1`] = `"cryptoLib"`;

exports[`interpreter default scripts crypto.src 2`] = `"cryptoLib"`;

exports[`interpreter default scripts crypto.src 3`] = `
"Waiting for beacon frame (BSSID: bssid-test-uuid)
Sending Authentication Request (Open system) [ACK]
Authentication succesful
Sending Association Request
Association succesful :-)"
`;

exports[`interpreter default scripts crypto.src 4`] = `"0/200"`;

exports[`interpreter default scripts crypto.src 5`] = `"0/200"`;

exports[`interpreter default scripts crypto.src 6`] = `"533/200"`;

exports[`interpreter default scripts crypto.src 7`] = `"null"`;

exports[`interpreter default scripts crypto.src 8`] = `"0"`;

exports[`interpreter default scripts crypto.src 9`] = `"null"`;

exports[`interpreter default scripts crypto.src 10`] = `"shallnotpass"`;

exports[`interpreter default scripts crypto.src 11`] = `"null"`;

exports[`interpreter default scripts fs.src 1`] = `"Before deletion: "`;

exports[`interpreter default scripts fs.src 2`] = `"object: file (file)"`;

exports[`interpreter default scripts fs.src 3`] = `"is folder:1 (number)"`;

exports[`interpreter default scripts fs.src 4`] = `"parent: file (file)"`;

exports[`interpreter default scripts fs.src 5`] = `"path: /root (string)"`;

exports[`interpreter default scripts fs.src 6`] = `"After deletion (old ref): "`;

exports[`interpreter default scripts fs.src 7`] = `"object: file (file)"`;

exports[`interpreter default scripts fs.src 8`] = `"is folder:1 (number)"`;

exports[`interpreter default scripts fs.src 9`] = `"parent:  (null)"`;

exports[`interpreter default scripts fs.src 10`] = `"path:  (null)"`;

exports[`interpreter default scripts fs.src 11`] = `"After deletion (new ref): "`;

exports[`interpreter default scripts fs.src 12`] = `"object:  (null)"`;

exports[`interpreter default scripts generics.src 1`] = `"null"`;

exports[`interpreter default scripts generics.src 2`] = `"098f6bcd4621d373cade4e832627b4f6"`;

exports[`interpreter default scripts generics.src 3`] = `"0"`;

exports[`interpreter default scripts generics.src 4`] = `"172.57.134.56"`;

exports[`interpreter default scripts generics.src 5`] = `
"Domain name: hampbell.com
Administrative contact: Brenald Ameen
Email address: e129895@hampbell.com
Phone: 123456891"
`;

exports[`interpreter default scripts generics.src 6`] = `"1"`;

exports[`interpreter default scripts generics.src 7`] = `"1"`;

exports[`interpreter default scripts generics.src 8`] = `"Unknown info"`;

exports[`interpreter default scripts generics.src 9`] = `"/home"`;

exports[`interpreter default scripts generics.src 10`] = `"/home/test"`;

exports[`interpreter default scripts generics.src 11`] = `"/home/test/myprogram"`;

exports[`interpreter default scripts generics.src 12`] = `"test"`;

exports[`interpreter default scripts generics.src 13`] = `"test@tierriso.tv"`;

exports[`interpreter default scripts generics.src 14`] = `"s1jbeulm"`;

exports[`interpreter default scripts generics.src 15`] = `"clearing screen"`;

exports[`interpreter default scripts generics.src 16`] = `"null"`;

exports[`interpreter default scripts generics.src 17`] = `"/home/test/myprogram"`;

exports[`interpreter default scripts generics.src 18`] = `
"COL1    COL2    COL3    
one     two     three   
thing1  thing2  thing3  "
`;

exports[`interpreter default scripts generics.src 19`] = `"test"`;

exports[`interpreter default scripts generics.src 20`] = `"test"`;

exports[`interpreter default scripts generics.src 21`] = `"test"`;

exports[`interpreter default scripts generics.src 22`] = `"test"`;

exports[`interpreter default scripts generics.src 23`] = `"test"`;

exports[`interpreter default scripts generics.src 24`] = `""`;

exports[`interpreter default scripts generics.src 25`] = `"null"`;

exports[`interpreter default scripts generics.src 26`] = `"number"`;

exports[`interpreter default scripts generics.src 27`] = `"string"`;

exports[`interpreter default scripts generics.src 28`] = `"number"`;

exports[`interpreter default scripts generics.src 29`] = `"function"`;

exports[`interpreter default scripts generics.src 30`] = `"null"`;

exports[`interpreter default scripts generics.src 31`] = `
"<b>Usage: ls [opt:-l, -a, -la] [opt: path]</b>
It can be called without parameters."
`;

exports[`interpreter default scripts generics.src 32`] = `"Unknown info"`;

exports[`interpreter default scripts generics.src 33`] = `"null"`;

exports[`interpreter default scripts get-custom-object.src 1`] = `"1"`;

exports[`interpreter default scripts get-custom-object.src 2`] = `"file"`;

exports[`interpreter default scripts get-custom-object.src 3`] = `""`;

exports[`interpreter default scripts get-custom-object.src 4`] = `"hello world"`;

exports[`interpreter default scripts isa.src 1`] = `"1"`;

exports[`interpreter default scripts isa.src 2`] = `"0"`;

exports[`interpreter default scripts isa.src 3`] = `"0"`;

exports[`interpreter default scripts isa.src 4`] = `"1"`;

exports[`interpreter default scripts isa.src 5`] = `"0"`;

exports[`interpreter default scripts isa.src 6`] = `"0"`;

exports[`interpreter default scripts ls.src 1`] = `
"Desktop    
Downloads  
Config     
myprogram  "
`;

exports[`interpreter default scripts meta-mail.src 1`] = `"MetaMail"`;

exports[`interpreter default scripts meta-mail.src 2`] = `"MetaMail"`;

exports[`interpreter default scripts meta-mail.src 3`] = `"[]"`;

exports[`interpreter default scripts meta-mail.src 4`] = `"Mail not found"`;

exports[`interpreter default scripts meta-mail.src 5`] = `"1"`;

exports[`interpreter default scripts meta-mail.src 6`] = `
"[\\"

MailID: c18554cf-1c6d-4ee4-ba86-7e6e78016002
From: test@test.org
Subject: hello
dude\\"]"
`;

exports[`interpreter default scripts meta-mail.src 7`] = `"c18554cf-1c6d-4ee4-ba86-7e6e78016002"`;

exports[`interpreter default scripts meta-mail.src 8`] = `
"
From: test@test.org
Subject: hello
dude

"
`;

exports[`interpreter default scripts meta-mail.src 9`] = `"1"`;

exports[`interpreter default scripts meta-mail.src 10`] = `"[]"`;

exports[`interpreter default scripts metaxploit.src 1`] = `"MetaxploitLib"`;

exports[`interpreter default scripts metaxploit.src 2`] = `"MetaxploitLib"`;

exports[`interpreter default scripts metaxploit.src 3`] = `"null"`;

exports[`interpreter default scripts metaxploit.src 4`] = `"null"`;

exports[`interpreter default scripts metaxploit.src 5`] = `"error: service rshelld is not running"`;

exports[`interpreter default scripts metaxploit.src 6`] = `"MetaLib"`;

exports[`interpreter default scripts metaxploit.src 7`] = `"MetaLib"`;

exports[`interpreter default scripts metaxploit.src 8`] = `"[\\"0x65E8E4EC\\", \\"0x3EB84DBB\\", \\"0x69498C3F\\", \\"0x1823E086\\"]"`;

exports[`interpreter default scripts metaxploit.src 9`] = `
"decompiling source...
searching unsecure values...
Unsafe check: loop in array <b>Tuiteralse</b>. Buffer overflow.
* Using namespace <b>init.so</b> compiled at version >= <b>1.0.0</b>
* Checking registered users equal to 3.
* Checking an active user.
* Checking root active user.

Unsafe check: loop in array <b>Tionpoint_onclosesour</b>. Buffer overflow.
* Checking root active user.

Unsafe check: string copy in <b>Textsizede</b>. Buffer overflow.
* Using namespace <b>net.so</b> compiled at version >= <b>1.0.0</b>
* Checking root active user.

Unsafe check: loop in array <b>Olor_buttoncloseinoverlayon</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0</b>
* Checking an active user.
* Checking root active user.

Unsafe check: string copy in <b>Eady_defieldin</b>. Buffer overflow.

Unsafe check: loop in array <b>Rhsa</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0</b>
* Checking registered users equal to 1.
* Checking an active user.
* Checking root active user.
"
`;

exports[`interpreter default scripts metaxploit.src 10`] = `"crypto.so"`;

exports[`interpreter default scripts metaxploit.src 11`] = `"1.0.0"`;

exports[`interpreter default scripts metaxploit.src 12`] = `"[\\"0x65E8E4EC\\", \\"0x3EB84DBB\\", \\"0x69498C3F\\", \\"0x1823E086\\"]"`;

exports[`interpreter default scripts metaxploit.src 13`] = `
"decompiling source...
searching unsecure values...
Unsafe check: loop in array <b>Tuiteralse</b>. Buffer overflow.
* Using namespace <b>init.so</b> compiled at version >= <b>1.0.0</b>
* Checking registered users equal to 3.
* Checking an active user.
* Checking root active user.

Unsafe check: loop in array <b>Tionpoint_onclosesour</b>. Buffer overflow.
* Checking root active user.

Unsafe check: string copy in <b>Textsizede</b>. Buffer overflow.
* Using namespace <b>net.so</b> compiled at version >= <b>1.0.0</b>
* Checking root active user.

Unsafe check: loop in array <b>Olor_buttoncloseinoverlayon</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0</b>
* Checking an active user.
* Checking root active user.

Unsafe check: string copy in <b>Eady_defieldin</b>. Buffer overflow.

Unsafe check: loop in array <b>Rhsa</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0</b>
* Checking registered users equal to 1.
* Checking an active user.
* Checking root active user.
"
`;

exports[`interpreter default scripts metaxploit.src 14`] = `
"
Starting attack...success!
Privileges obtained from user: test"
`;

exports[`interpreter default scripts metaxploit.src 15`] = `"file"`;

exports[`interpreter default scripts metaxploit.src 16`] = `
"
Starting attack...success!
Privileges obtained from user: test"
`;

exports[`interpreter default scripts metaxploit.src 17`] = `"file"`;

exports[`interpreter default scripts metaxploit.src 18`] = `"NetSession"`;

exports[`interpreter default scripts metaxploit.src 19`] = `"NetSession"`;

exports[`interpreter default scripts metaxploit.src 20`] = `"MetaLib"`;

exports[`interpreter default scripts metaxploit.src 21`] = `"MetaLib"`;

exports[`interpreter default scripts metaxploit.src 22`] = `"libssh.so"`;

exports[`interpreter default scripts metaxploit.src 23`] = `"1.0.0"`;

exports[`interpreter default scripts metaxploit.src 24`] = `"[\\"0x2C35E48D\\", \\"0x2EC47F98\\", \\"0x6AB1D4F3\\", \\"0x7D20AD77\\", \\"0x591A7117\\"]"`;

exports[`interpreter default scripts metaxploit.src 25`] = `
"decompiling source...
searching unsecure values...
Unsafe check: string copy in <b>Animator3</b>. Buffer overflow.

Unsafe check: string copy in <b>Thsbuild_flush_bloc</b>. Buffer overflow.

Unsafe check: string copy in <b>Treelitextsiintof</b>. Buffer overflow.

Unsafe check: loop in array <b>Lush_s</b>. Buffer overflow.
* Using namespace <b>kernel_module.so</b> compiled at version >= <b>1.0.0</b>
* Checking registered users equal to 1.
* 1 port forwarding configured from router to the target computer.
"
`;

exports[`interpreter default scripts metaxploit.src 26`] = `
"
Starting attack...success!
Computer obtained with credentials from user: test"
`;

exports[`interpreter default scripts metaxploit.src 27`] = `"computer"`;

exports[`interpreter default scripts metaxploit.src 28`] = `"kernel_router.so"`;

exports[`interpreter default scripts metaxploit.src 29`] = `"1.0.0"`;

exports[`interpreter default scripts metaxploit.src 30`] = `"[\\"0x243A0F59\\", \\"0x2F618803\\", \\"0x45917FA3\\", \\"0x2542555D\\"]"`;

exports[`interpreter default scripts metaxploit.src 31`] = `
"decompiling source...
searching unsecure values...
Unsafe check: loop in array <b>_lit++</b>. Buffer overflow.

Unsafe check: loop in array <b>Atelend_bit</b>. Buffer overflow.

Unsafe check: string copy in <b>_bits</b>. Buffer overflow.

Unsafe check: loop in array <b>Ush_bloc</b>. Buffer overflow.

Unsafe check: loop in array <b>Typebott</b>. Buffer overflow.
"
`;

exports[`interpreter default scripts metaxploit.src 32`] = `
"
Starting attack...success!
Privileges obtained from user: otisactiv"
`;

exports[`interpreter default scripts metaxploit.src 33`] = `"shell"`;

exports[`interpreter default scripts metaxploit.src 34`] = `"14"`;

exports[`interpreter default scripts metaxploit.src 35`] = `"6"`;

exports[`interpreter default scripts metaxploit.src 36`] = `"2"`;

exports[`interpreter default scripts metaxploit.src 37`] = `"1"`;

exports[`interpreter default scripts metaxploit.src 38`] = `"1"`;

exports[`interpreter default scripts mkdir.src 1`] = `"0"`;

exports[`interpreter default scripts router.src 1`] = `"router"`;

exports[`interpreter default scripts router.src 2`] = `"router"`;

exports[`interpreter default scripts router.src 3`] = `"router"`;

exports[`interpreter default scripts router.src 4`] = `"router"`;

exports[`interpreter default scripts router.src 5`] = `"142.32.54.56"`;

exports[`interpreter default scripts router.src 6`] = `"192.168.0.1"`;

exports[`interpreter default scripts router.src 7`] = `"port"`;

exports[`interpreter default scripts router.src 8`] = `"ssh 1.0.0"`;

exports[`interpreter default scripts router.src 9`] = `"[port, port, port, port, port, port, port]"`;

exports[`interpreter default scripts router.src 10`] = `"1.0.0"`;

exports[`interpreter default scripts scripts.src 1`] = `
"Desktop    
Downloads  
Config     "
`;

exports[`interpreter default scripts scripts.src 2`] = `
"-r-x--x--x  root  root  345   00:00  build             
-r-x--x--x  root  root  262   00:00  cat               
-r-x--x--x  root  root  848   00:00  cp                
-r-x--x--x  root  root  261   00:00  kill              
-r-x--x--x  root  root  1014  00:00  ls                
-r-x--x--x  root  root  505   00:00  mkdir             
-r-x--x--x  root  root  918   00:00  mv                
-r-x--x--x  root  root  89    00:00  ps                
-r-x--x--x  root  root  23    00:00  pwd               
-r-x--x--x  root  root  401   00:00  rm                
-r-x--x--x  root  root  637   00:00  sudo              
-r-x--x--x  root  root  373   00:00  touch             
-r-x--x--x  root  root  24    00:00  whoami            
-r-x--x--x  root  root  415   00:00  aircrack          
-r-x--x--x  root  root  249   00:00  aireplay          
-r-x--x--x  root  root  470   00:00  airmon            
-r-x--x--x  root  root  2157  00:00  apt-get           
-r-x--x--x  root  root  343   00:00  ftp               
-r-x--x--x  root  root  734   00:00  ifconfig          
-r-x--x--x  root  root  298   00:00  iwconfig          
-r-x--x--x  root  root  388   00:00  iwlist            
-r-x--x--x  root  root  779   00:00  nmap              
-r-x--x--x  root  root  126   00:00  nslookup          
-r-x--x--x  root  root  196   00:00  ping              
-r-x--x--x  root  root  612   00:00  rshell-interface  
-r-x--x--x  root  root  442   00:00  smtp-user-list    
-r-x--x--x  root  root  354   00:00  sniffer           
-r-x--x--x  root  root  394   00:00  ssh               
-r-x--x--x  root  root  113   00:00  whois             
-r-x--x--x  root  root  284   00:00  chgrp             
-r-x--x--x  root  root  288   00:00  chmod             
-r-x--x--x  root  root  284   00:00  chown             
-r-x--x--x  root  root  233   00:00  groupadd          
-r-x--x--x  root  root  236   00:00  groupdel          
-r-x--x--x  root  root  159   00:00  groups            
-r-x--x--x  root  root  266   00:00  passwd            
-r-x--x--x  root  root  268   00:00  useradd           
-r-x--x--x  root  root  269   00:00  userdel           
-r-x--x--x  root  root  998   00:00  decipher          
-r-x--x--x  root  root  693   00:00  scanlib           
-r-x--x--x  root  root  519   00:00  scanrouter        "
`;

exports[`interpreter default scripts scripts.src 3`] = `
"USER  PID   CPU   MEM    COMMAND      
root  3060  6.9   8.10   kernel_task  
root  1588  11.6  24.27  Xorg         
test  3288  14.7  22.26  myprogram    "
`;

exports[`interpreter default scripts scripts.src 4`] = `"/root"`;

exports[`interpreter default scripts scripts.src 5`] = `
"
Connected to Wi-Fi:
Essid: anildx
Bssid: 44:8E:73:9E:F2:63
----------------
Public IP: 142.32.54.56
Local IP: 192.168.195.13
Gateway: 192.168.0.1
"
`;

exports[`interpreter default scripts service.src 1`] = `"service"`;

exports[`interpreter default scripts service.src 2`] = `"service"`;

exports[`interpreter default scripts service.src 3`] = `"Denied. Only root user can install this service."`;

exports[`interpreter default scripts service.src 4`] = `"null"`;

exports[`interpreter default scripts service.src 5`] = `"Denied. Only root user can install this service."`;

exports[`interpreter default scripts shell.src 1`] = `"shell"`;

exports[`interpreter default scripts shell.src 2`] = `"computer"`;

exports[`interpreter default scripts shell.src 3`] = `"computer"`;

exports[`interpreter default scripts shell.src 4`] = `"[port]"`;

exports[`interpreter default scripts shell.src 5`] = `"Can't create folder /root/foo. Permission denied"`;

exports[`interpreter default scripts shell.src 6`] = `"1"`;

exports[`interpreter default scripts shell.src 7`] = `"file"`;

exports[`interpreter default scripts shell.src 8`] = `"1"`;

exports[`interpreter default scripts shell.src 9`] = `"Can't create file /root/foo-file. Permission denied"`;

exports[`interpreter default scripts shell.src 10`] = `"1"`;

exports[`interpreter default scripts shell.src 11`] = `"0"`;

exports[`interpreter default scripts shell.src 12`] = `
"USER PID CPU MEM COMMAND
root 6938 13.8 20.39 kernel_task
root 8513 2.3 20.73 Xorg
test 3604 3.1 26.25 myprogram"
`;

exports[`interpreter default scripts shell.src 13`] = `"wlan0 ggrdk4zo true"`;

exports[`interpreter default scripts shell.src 14`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 15`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 16`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 17`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 18`] = `""`;

exports[`interpreter default scripts shell.src 19`] = `"Denied. Only root user can execute this command."`;

exports[`interpreter default scripts shell.src 20`] = `"0"`;

exports[`interpreter default scripts shell.src 21`] = `"[\\"bssid-test-uuid 52% essid-test-uuid\\", \\"CE:A4:DD:DF:73:29 61% vingoftw\\", \\"42:D6:90:7C:BF:71 84% aptain117\\", \\"09:2F:46:FB:32:E9 80% f999\\"]"`;

exports[`interpreter default scripts shell.src 22`] = `"1"`;

exports[`interpreter default scripts shell.src 23`] = `"connect_ethernet: Network device not found"`;

exports[`interpreter default scripts shell.src 24`] = `"192.168.0.10"`;

exports[`interpreter default scripts shell.src 25`] = `"WIFI"`;

exports[`interpreter default scripts shell.src 26`] = `"192.168.93.53"`;

exports[`interpreter default scripts shell.src 27`] = `"20.118.3.131"`;

exports[`interpreter default scripts shell.src 28`] = `"file"`;

exports[`interpreter default scripts shell.src 29`] = `"file"`;

exports[`interpreter default scripts shell.src 30`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 31`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 32`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 33`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 34`] = `"/root"`;

exports[`interpreter default scripts shell.src 35`] = `"file"`;

exports[`interpreter default scripts shell.src 36`] = `"root"`;

exports[`interpreter default scripts shell.src 37`] = `"null"`;

exports[`interpreter default scripts shell.src 38`] = `"can't open /root Binary file."`;

exports[`interpreter default scripts shell.src 39`] = `"1"`;

exports[`interpreter default scripts shell.src 40`] = `"1"`;

exports[`interpreter default scripts shell.src 41`] = `"0"`;

exports[`interpreter default scripts shell.src 42`] = `"permission denied"`;

exports[`interpreter default scripts shell.src 43`] = `"[file, file, file, file]"`;

exports[`interpreter default scripts shell.src 44`] = `"[]"`;

exports[`interpreter default scripts shell.src 45`] = `"root"`;

exports[`interpreter default scripts shell.src 46`] = `"Permission denied"`;

exports[`interpreter default scripts shell.src 47`] = `"root"`;

exports[`interpreter default scripts shell.src 48`] = `"Permission denied"`;

exports[`interpreter default scripts shell.src 49`] = `"0"`;

exports[`interpreter default scripts shell.src 50`] = `"0"`;

exports[`interpreter default scripts shell.src 51`] = `"computer"`;

exports[`interpreter default scripts shell.src 52`] = `"0"`;

exports[`interpreter default scripts shell.src 53`] = `"1"`;

exports[`interpreter default scripts shell.src 54`] = `"file"`;

exports[`interpreter default scripts shell.src 55`] = `"1"`;

exports[`interpreter default scripts shell.src 56`] = `"file"`;

exports[`interpreter default scripts shell.src 57`] = `""`;

exports[`interpreter default scripts shell.src 58`] = `"123"`;

exports[`interpreter default scripts shell.src 59`] = `"1"`;

exports[`interpreter default scripts shell.src 60`] = `"1"`;

exports[`interpreter default scripts shell.src 61`] = `"testg"`;

exports[`interpreter default scripts shell.src 62`] = `"1"`;

exports[`interpreter default scripts shell.src 63`] = `"1"`;

exports[`interpreter default scripts shell.src 64`] = `"1"`;

exports[`interpreter default scripts shell.src 65`] = `"1"`;

exports[`interpreter default scripts shell.src 66`] = `"1"`;

exports[`interpreter default scripts shell.src 67`] = `"file"`;

exports[`interpreter default scripts shell.src 68`] = `""`;

exports[`interpreter default scripts shell.src 69`] = `"drwxr--rwx"`;

exports[`interpreter default scripts shell.src 70`] = `"1"`;

exports[`interpreter default scripts shell.src 71`] = `"1"`;

exports[`interpreter default scripts shell.src 72`] = `"i am admin"`;

exports[`interpreter default scripts shell.src 73`] = `"root"`;

exports[`interpreter default scripts shell.src 74`] = `""`;

exports[`interpreter default scripts shell.src 75`] = `"test"`;

exports[`interpreter default scripts shell.src 76`] = `"root"`;

exports[`interpreter default scripts shell.src 77`] = `""`;

exports[`interpreter default scripts shell.src 78`] = `"test-here"`;

exports[`interpreter default scripts shell.src 79`] = `"/"`;

exports[`interpreter default scripts shell.src 80`] = `"entooxi"`;

exports[`interpreter default scripts shell.src 81`] = `"shell"`;

exports[`interpreter default scripts shell.src 82`] = `"shell"`;

exports[`interpreter default scripts shell-self-launch.src 1`] = `"launch number 0"`;

exports[`interpreter default scripts shell-self-launch.src 2`] = `"launch number 1"`;

exports[`interpreter default scripts shell-self-launch.src 3`] = `"launch number 2"`;

exports[`interpreter default scripts shell-self-launch.src 4`] = `"launch number 3"`;

exports[`interpreter default scripts shell-self-launch.src 5`] = `"launch number 4"`;

exports[`interpreter default scripts shell-self-launch.src 6`] = `"launch number 5"`;

exports[`interpreter default scripts shell-self-launch.src 7`] = `"finished self launch"`;

exports[`interpreter default scripts sub-wallet.src 1`] = `"subwallet"`;

exports[`interpreter default scripts sub-wallet.src 2`] = `"subwallet"`;

exports[`interpreter default scripts sub-wallet.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 6`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 7`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 8`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 9`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 10`] = `"Not yet supported"`;

exports[`interpreter default scripts sub-wallet.src 11`] = `"Not yet supported"`;

exports[`interpreter default scripts test-lib.src 1`] = `"[{\\"device\\":computer,\\"program\\":file,\\"currentPath\\":file}]"`;

exports[`interpreter default scripts test-lib.src 2`] = `"router"`;

exports[`interpreter default scripts test-lib.src 3`] = `"{\\"192.168.33.152\\":computer,\\"192.168.42.250\\":computer,\\"192.168.64.69\\":computer,\\"192.168.70.249\\":computer,\\"192.168.207.138\\":computer,\\"192.168.70.55\\":computer,\\"192.168.28.205\\":computer,\\"192.168.231.43\\":computer}"`;

exports[`interpreter default scripts test-lib.src 4`] = `"shell"`;

exports[`interpreter default scripts test-lib.src 5`] = `"shell"`;

exports[`interpreter default scripts test-lib.src 6`] = `"computer"`;

exports[`interpreter default scripts test-lib.src 7`] = `"at internal:0:0"`;

exports[`interpreter default scripts test-lib.src 8`] = `"test-lib.src:27:23"`;

exports[`interpreter default scripts test-lib.src 9`] = `"test-lib.src:34:3"`;

exports[`interpreter default scripts test-lib.src 10`] = `"test-lib.src:38:3"`;

exports[`interpreter default scripts test-lib.src 11`] = `"test-lib.src:41:1"`;

exports[`interpreter default scripts wallet.src 1`] = `"wallet"`;

exports[`interpreter default scripts wallet.src 2`] = `"wallet"`;

exports[`interpreter default scripts wallet.src 3`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 4`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 5`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 6`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 7`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 8`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 9`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 10`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 11`] = `"Not yet supported"`;

exports[`interpreter default scripts wallet.src 12`] = `"Not yet supported"`;
